content/hi/docs/concepts/overview/components.md

```diff
@@ -0,0 +1,130 @@
+
+### kube-controller-manager
+
+{{< glossary_definition term_id="kube-controller-manager" length="all" >}}
+
+इन नियंत्रकों के कुछ प्रकार हैं:
+
+  * नोड नियंत्रक: नोड्स विफल होने पर सूचित करने और प्रतिक्रिया देने के लिए जिम्मेदार।
+  * जॉब नियंत्रक: जॉब ऑब्जेक्ट्स को देखता है जो एक बार के कार्यों का प्रतिनिधित्व करता है, फिर उन कार्यों को
+    पूरा करने के लिए पॉड्स बनाता है।
+  * एंडपॉइंटस्लाइस नियंत्रक: एंडपॉइंटस्लाइस ऑब्जेक्ट्स को पॉप्युलेट करता है (सेवाओं और पॉड्स के बीच एक लिंक प्रदान करने के लिए)।
+  * सर्विस अकाउंट नियंत्रक: नए नेमस्पेस के लिए डिफॉल्ट सर्विस अकाउंट बनाएँ।
+
+### cloud-controller-manager
+
+{{< glossary_definition term_id="cloud-controller-manager" length="short" >}}
+
+क्लाउड-नियंत्रक-प्रबंधक केवल उन नियंत्रकों को चलाता है जो आपके क्लाउड प्रदाता के लिए विशिष्ट हैं।
+यदि आप अपने स्वयं के परिसर में कुबेरनेट चला रहे हैं, या अपने अंदर सीखने के माहौल में हैं
+अपने पीसी, क्लस्टर में क्लाउड कंट्रोलर मैनेजर नहीं है।
+
+क्यूब-कंट्रोलर-मैनेजर की तरह, क्लाउड-कंट्रोलर-मैनेजर कई तार्किक रूप से जोड़ता है
+स्वतंत्र नियंत्रण एक एकल बाइनरी में लूप करता है जिसे आप एकल प्रक्रिया के रूप में चलाते हैं। तुम कर सकते हो
+प्रदर्शन में सुधार करने या विफलताओं को सहन करने में सहायता के लिए क्षैतिज रूप से स्केल करें (एक से अधिक प्रतिलिपि चलाएं)।
+
+निम्नलिखित नियंत्रकों में क्लाउड प्रदाता निर्भरताएँ हो सकती हैं:
+
+  * नोड नियंत्रक: क्लाउड प्रदाता की जांच के लिए यह निर्धारित करने के लिए कि क्लाउड में नोड को प्रतिक्रिया देना बंद करने के बाद हटा दिया गया है या नहीं
+  * मार्ग नियंत्रक: अंतर्निहित क्लाउड अवसंरचना में मार्ग स्थापित करने के लिए
+  * सर्विस कंट्रोलर: क्लाउड प्रोवाइडर लोड बैलेंसर्स को बनाने, अपडेट करने और हटाने के लिए
+
+## Node Components
+
+नोड घटक प्रत्येक नोड पर चलते हैं, रनिंग पॉड्स को बनाए रखते हैं और कुबेरनेट्स रनटाइम वातावरण प्रदान करते हैं।
+
+## kubelet
+
+{{< glossary_definition term_id="kubelet" length="all" >}}
+
+### kube-proxy
+
+{{< glossary_definition term_id="kube-proxy" length="all" >}}
+
+### Container runtime
+
+{{< glossary_definition term_id="container-runtime" length="all" >}}
+
+## Addons
+
+एडॉन्स क्लस्टर सुविधाओं को लागू करने के लिए कुबेरनेट्स संसाधनों ({{< glossary_tooltip term_id="daemonset" >}},
+{{< glossary_tooltip term_id="deployment" >}}, आदि)
+ का उपयोग करते हैं। क्योंकि ये क्लस्टर-स्तर की सुविधाएँ प्रदान 
+ कर रहे हैं, ऐड-ऑन के लिए नामित संसाधन `क्यूब-सिस्टम` नेमस्पेस के अंतर्गत आते हैं।
+
+चयनित योजकों का वर्णन नीचे किया गया है; उपलब्ध एडॉन्स की विस्तृत सूची के लिए, 
+कृपया [एडॉन्स](/docs/concepts/cluster-administration/addons/) देखें।
+
+### DNS
+
+जबकि अन्य एडोनों की सख्ती से आवश्यकता नहीं है, सभी कुबेरनेट समूहों में [क्लस्टर डीएनएस](/docs/concepts/services-networking/dns-pod-service/) होना चाहिए, क्योंकि कई उदाहरण इस पर भरोसा करते हैं।
+
+क्लस्टर डीएनएस एक डीएनएस सर्वर है, जो आपके वातावरण में अन्य डीएनएस सर्वर के अलावा कुबेरनेट्स सेवाओं के लिए डीएनएस रिकॉर्ड प्रदान करता है।
+
+कुबेरनेट्स द्वारा शुरू किए गए कंटेनर स्वचालित रूप से इस डीएनएस सर्वर को अपनी डीएनएस खोजों में शामिल करते हैं।
```

Comment:
I think `खोज सूची` better conveys the word searches in this case, thoughts, WDYT @NitishKumar06?
```suggestion
कुबेरनेट्स द्वारा शुरू किए गए कंटेनर स्वचालित रूप से इस DNS सर्वर को अपनी DNS खोज सूची में शामिल करते हैं।
```